library/src/main/java/com/datatorrent/lib/util/StorageAgentKeyValueStore.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.util;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Interface for KeyValue store
+ * 
+ */
+public interface StorageAgentKeyValueStore
+{
+
+  /**
+   * Connects to the service.
+   *
+   * @throws IOException
+   */
+  public void connect() throws IOException;
+
+  /**
+   * Disconnects from the service.
+   *
+   * @throws IOException
+   */
+  public void disconnect() throws IOException;
+
+  /**
+   *
+   * @return returns whether the service is connected.
+   */
+  public boolean isConnected();
+
+  /**
+   * Gets the value given the key.
+   *
+   * @param key
+   * @return the value
+   */
+  public Object get(Object key);
+
+  /**
+   * Sets the key with the value in the store.
+   *
+   * @param key
+   * @param value
+   */
+  public void put(Object key, Object value);
+
+  /**
+   * Removes the key and the value given the key
+   * 
+   * @param key
+   */
+  public void remove(Object key);
+
+  /**
+   * Get all the keys associated with key
+   * 
+   * @param key
+   * @return the list of all associated keys
+   */
+  public List<String> getKeys(Object key);
+
+  /**
+   * Set table/region name of store
+   * 
+   * @param tableName
+   */
+  public void setTableName(String tableName);
```

Comment:
I thought it would be common for any key value store service as most of them have notion  of table, region to denote to particular storage instance.